src/librustc_driver/lib.rs

```diff
@@ -1136,6 +1136,16 @@ fn usage(verbose: bool, include_unstable_options: bool) {
              verbose_help);
 }
 
+fn print_wall_help() {
+    println!("
+The flag -Wall does not exist in rustc. Most useful lints are enabled by default.
+Use `rustc -W help` to see all available lints. The most used lints that are not
+enabled by default covered by -Wunused; however, the best practice is to put
+warning settings in the crate root using `#![warn(unused)]` instead of using
+the command line flag directly.
```

Comment:
I'm fine with the feature. I would like to have a native speaker take a look over this. cc @rust-lang/docs 

Maybe something closer to:

```
The flag `-Wall` does not exist in `rustc`. Most useful lints are enabled by default.
Use `rustc -W help` to see all available lints. The most used lints that are not
enabled by default are covered by `-Wunused`; however, the best practice is to
put warning settings in the crate root using `#![warn(LINT_NAME)]` instead of using
the command line flag directly.
```

Reply:
`unused` turns on _half_ of the allowed-by-default lints that I've ever seen used:
- `unused-qualifications`
- `unused-results`
- `unused-import-braces`
- `unused-extern-crates`

The others that I've personally seen used are:
- `missing-docs`
- `trivial-casts`
- `trivial-numeric-casts`
- `unreachable-pub`

And given that a LOT of crates in the ecosystem have `#![warn(unused)]`, I thought it was the most relevant one to mention.

Reply:
Actually, looking at the `-W help` output, `-Wunused` only turns on `unused-extern-crates`... I thought it did all the `unused-*` lints?
I might have to go step-up my `#![warn(unused-*)]` lines in my projects.

Reply:
> Actually, looking at the `-W help` output, `-Wunused` only turns on `unused-extern-crates`

taken from `rustc -W help`

```
Lint groups provided by rustc:

                   name  sub-lints
                   ----  ---------
               warnings  all lints that are set to issue warnings
              bad-style  non-camel-case-types, non-snake-case, non-upper-case-globals
    future-incompatible  private-in-public, pub-use-of-private-extern-crate, patterns-in-fns-without-body, safe-extern-statics, invalid-type-param-default, legacy-directory-ownership, legacy-imports, legacy-constructor-visibility, resolve-trait-on-defaulted-unit, missing-fragment-specifier, illegal-floating-point-literal-pattern, anonymous-parameters, parenthesized-params-in-types-and-modules, late-bound-lifetime-arguments, safe-packed-borrows, incoherent-fundamental-impls, coerce-never, tyvar-behind-raw-pointer
                 unused  unused-imports, unused-variables, unused-assignments, dead-code, unused-mut, unreachable-code, unreachable-patterns, unused-must-use, unused-unsafe, path-statements, unused-attributes, unused-macros, unused-allocation, unused-doc-comment, unused-extern-crates, unused-features, unused-parens


Compiler plugins can provide additional lints and lint groups. To see a listing of these, re-run `rustc -W help` with a crate filename.
```